src/actions/appActions.js

```diff
@@ -187,11 +189,21 @@ export const initAppAndRedirectAction = () => {
       const user = storageData?.user?.user ?? {};
       dispatch({ type: SET_USER, payload: user });
 
+      const { data }: any = await httpRequest
+        .get(WORLD_TIME_API, {
+          timeout: API_REQUEST_TIMEOUT * 6,
+        })
+        .catch((error) => {
+          logBreadcrumb('WorldTimeApi', 'Failed: world time api', { error });
+          return { data: null };
+        });
+
       const { failedAttempts = {}, pinAttempt = {} } = get(storageData, 'pinAttempt', {});
-      const { numberOfFailedAttempts = 0, date = new Date() } = failedAttempts;
+      const { numberOfFailedAttempts = 0, date = data?.datetime || new Date() } = failedAttempts;
       const { pinAttemptsCount = 0 } = pinAttempt;
 
-      if (new Date(date)?.toDateString() === new Date()?.toDateString()) {
+      const today = data?.datetime ? new Date(data.datetime)?.toDateString() : new Date()?.toDateString();
```

Comment:
did we need the ? on the `new Date()?.toDateString()`?

Reply:
did we need the ? on the new Date()?.toDateString()?
Not mandatory, I'll remove it. 

I saw you added the flag to isLoading flag, does it just display a spinner?

https://github.com/pillarwallet/pillarwallet/assets/82652040/584cb0a9-7178-4c7d-b5b0-bd68532e7811

Reply:
@lalitkumarrock thanks. Does pressing any digits of the keypad have any effect here? Just trying to close all possible attack vectors

Reply:
At that time not working any keypad.

https://github.com/pillarwallet/pillarwallet/assets/82652040/d8cb08b0-9a55-4998-a36f-a65f07c66d0a